packages/blade/src/components/FileUpload/FileUpload.web.tsx

```diff
@@ -0,0 +1,413 @@
+
+  // In control mode attach a unique id to each file if not provided
+  useMemo(() => {
+    for (const file of selectedFiles) {
+      if (!file.id) {
+        file.id = `${new Date().getTime().toString()}${Math.floor(Math.random() * 1000000)}`;
+      }
+    }
+  }, [selectedFiles]);
+
+  const handleFilesChange = useCallback((inputFiles: BladeFileList) => {
+    setSelectedFiles((prevFiles) => {
+      if (prevFiles.length > 0) {
+        const allFiles = [...prevFiles, ...inputFiles];
+        return allFiles;
+      }
+
+      return inputFiles;
+    });
+  }, []);
+
+  const validateFiles = (inputFiles: BladeFileList, allFiles: BladeFileList): boolean => {
+    if (accept && inputFiles.some((file) => !isFileAccepted(file, accept))) {
+      setErrorMessage(`You provided an unsupported file type. Supported file types are: ${accept}`);
+      setInternalValidationState('error');
+      return true;
+    }
+
+    if (uploadType === 'single' && inputFiles.length > 1) {
+      setErrorMessage('You can upload only one file.');
+      setInternalValidationState('error');
+      return true;
+    }
+
+    if (maxCount && allFiles.length > maxCount) {
+      setErrorMessage(`You can't upload more than ${maxCount} files.`);
+      setInternalValidationState('error');
+      return true;
+    }
+
+    if (maxSize && inputFiles.some((file) => file.size > maxSize)) {
+      setErrorMessage('File size exceeded.');
+      setInternalValidationState('error');
+      return true;
+    }
+
+    setInternalValidationState('none');
+    setErrorMessage('');
+    return false;
+  };
+
+  const handleDragOver = (event: React.DragEvent): void => {
+    event.preventDefault();
+    setIsActive(true);
+  };
+
+  const handleDragLeave = (event: React.DragEvent): void => {
+    event.preventDefault();
+    setIsActive(false);
+  };
+
+  const handleDrop = (event: React.DragEvent): void => {
+    event.preventDefault();
+    setIsActive(false);
+
+    const droppedFiles = Array.from(event.dataTransfer.files);
+    const allFiles = selectedFiles.length > 0 ? [...selectedFiles, ...droppedFiles] : droppedFiles;
+
+    const hasValidationErrors = validateFiles(droppedFiles, allFiles);
+
+    if (!hasValidationErrors) {
+      handleFilesChange(droppedFiles);
+      onDrop?.({ name, fileList: allFiles });
+    }
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    const inputFiles = Array.from(event.target.files ?? []);
+    const allFiles = selectedFiles.length > 0 ? [...selectedFiles, ...inputFiles] : inputFiles;
+
+    const hasValidationErrors = validateFiles(inputFiles, allFiles);
+
+    if (!hasValidationErrors) {
+      handleFilesChange(inputFiles);
+      onChange?.({ name, fileList: allFiles });
+    }
+  };
+
+  return (
+    <BaseBox
+      display="flex"
+      flexDirection="column"
+      {...metaAttribute({ name: MetaConstants.FileUpload, testID })}
+      {...getStyledProps(styledProps)}
+      width="100%"
```

Comment:
Why are we enforcing width to be `100%`? Now if someone passes width as part of styledProps, that will never get applied since we are overriding it here

Reply:
My Bad! need to move this up.